generators/app/templates/infrastructure/src/utils/apiRestDataSource.js

```diff
@@ -0,0 +1,28 @@
+const { correlationManager } = require("@totalsoft/correlation");
+const NoCacheRESTDataSource = require("./noCacheRESTDataSource");
+const { TenantId, UserId, UserPassport } = require("../constants/customHttpHeaders");
+
+class ApiRESTDataSource extends NoCacheRESTDataSource {
+
+  willSendRequest(request) {
+    <%_ if(withMultiTenancy){ _%>
+    request.headers.set(TenantId, this.context.state?.jwtdata?.tid ?? "");
+    <%_}_%>
+    request.headers.set(UserPassport, this.context.state?.jwtdata ? JSON.stringify(this.context.state.jwtdata) : "");
+    request.headers.set(UserId, this.context.state?.jwtdata?.sub ?? "");
+
+    //TODO to be removed
+    if (this.context.token) {
+      request.headers.set("Authorization", `Bearer ${this.context.token}`);
+    }
+
+    if (this.context.request?.headers?.["accept-language"])
```

Comment:
try using variables/constants instead of using `this.context.[...]` many times.  
Also, aren't you already receiving the `request` as an argument here in `willSendRequest` method? Why you reading it again using `this`?

Reply:
the argumet of willSendRequest referrs to the outgoing request (to the rest API)
this.context.request referrs to the current (incoming) gql request